src/service.c

```diff
@@ -2920,7 +2920,7 @@ void service_notify_cb(uev_t *w, void *arg, int events)
 	buf[len] = 0;
 
 	/* systemd and s6, respectively.  The latter then closes the socket */
-	if (!strcmp(buf, "READY=1\n") || !strcmp(buf, "\n")) {
+	if (!strcmp(buf, "READY=1\n") || (strlen(buf) > 0 && buf[strlen(buf) - 1] == '\n')) {
```

Comment:
Nice catch!

Since `len` and `strlen(buf)` always are equal, we can simplify this line to:

```c
  if (!strcmp(buf, "READY=1\n") || (len > 0 && buf[len - 1] == '\n')) {
```

For even better error handling, we could move `len > 0` to the previous comparison:

```c
  if (len <= 0) {
    warn("...")
    return;
  }
```

Reply:
thanks for the suggestions :smile: